diff --git a/frontend_editing.module b/frontend_editing.module
index ab8d022..ceb3ec0 100644
--- a/frontend_editing.module
+++ b/frontend_editing.module
@@ -25,6 +25,7 @@ use Drupal\frontend_editing\Ajax\EntityPreviewCommand;
 use Drupal\frontend_editing\Ajax\ScrollTopCommand;
 use Drupal\frontend_editing\Form\ParagraphAddForm;
 use Drupal\frontend_editing\Form\ParagraphDeleteForm;
+use Drupal\node\NodeInterface;
 use Drupal\paragraphs\ParagraphInterface;
 
 /**
@@ -167,6 +168,11 @@ function frontend_editing_entity_view_alter(&$build, EntityInterface $entity, En
     'entity_reference',
   ];
   foreach (Element::getVisibleChildren($build) as $field_name) {
+
+    if ($field_name == '_layout_builder') {
+      _frontend_editing_process_layout_builder_sections($build['_layout_builder'], $entity, $allowed_field_types, $supported_entity_types, $config);
+    }
+
     if (!isset($build[$field_name]['#theme'])) {
       continue;
     }
@@ -440,6 +446,80 @@ function frontend_editing_entity_view_alter(&$build, EntityInterface $entity, En
   }
 }
 
+/**
+ * Process layout builder sections to add a wrapper to content of field_page_section.
+ * Add parent field view mode to render array of children.
+ */
+function _frontend_editing_process_layout_builder_sections(&$layout_builder, $entity, $allowed_field_types, $supported_entity_types, $config) {
+  foreach ($layout_builder as &$element) {
+    if (is_array($element)) {
+      if (isset($element['#plugin_id'])) {
+        $view_mode = $element['content'][0]['#view_mode'] ?? 'default';
+        $derivative_plugin_id = $element['#derivative_plugin_id'] ?? 'unknown';
+        $field_name = explode(':', $derivative_plugin_id);
+        $field_name = end($field_name);
+
+        if (!isset($element['content'][0]['#theme'])) {
+          continue;
+        }
+
+        if ($element['content'][0]['#theme'] != 'field') {
+          continue;
+        }
+
+        if (isset($element['content'][0]['#field_type']) && in_array($element['content'][0]['#field_type'], $allowed_field_types)) {
+
+          $settings = $element['content'][0]['#items']->getFieldDefinition()->getSettings();
+          // Only add the wrapper for entity reference fields that are targeting the
+          // supported for frontend editing entity types.
+          if (!in_array($settings['target_type'], array_keys($supported_entity_types), TRUE)) {
+            continue;
+          }
+
+          if (!empty($settings['handler_settings']['target_bundles']) && empty(array_intersect($settings['handler_settings']['target_bundles'], $supported_entity_types[$settings['target_type']]))) {
+            continue;
+          }
+
+          // If ajax content update is not allowed, no need to add field wrapper.
+          if (!$config->get('ajax_content_update')) {
+            continue;
+          }
+
+          // Check if the field is not excluded from frontend editing ajax content
+          // update.
+          if (!frontend_editing_add_field_wrapper($entity->getEntityTypeId(), $entity->bundle(), $field_name)) {
+            continue;
+          }
+
+          foreach (Element::getVisibleChildren($element['content'][0]) as $delta) {
+            $item = $element['content'][0][$delta];
+            if (!empty($item['#pre_render']) && isset($element['content'][0]['#view_mode'])) {
+              $element['content'][0][$delta]['#parent_field_view_mode'] = $element['content'][0]['#view_mode'];
+            }
+          }
+
+          $selector = $entity->getEntityTypeId() . '--' . $entity->id() . '--' . $field_name . '--' . $view_mode;
+
+          if (isset($element['content'])) {
+            $element['content']['#theme_wrappers'] = [
+              'container' => [
+                '#attributes' => [
+                  'data-frontend-editing' => $selector,
+                  'class' => [
+                    'frontend-editing-field-wrapper',
+                  ],
+                ],
+              ],
+            ];
+          }
+        }
+      }
+
+      _frontend_editing_process_layout_builder_sections($element, $entity, $allowed_field_types, $supported_entity_types, $config);
+    }
+  }
+}
+
 /**
  * Implements hook_theme().
  */
diff --git a/src/Controller/FrontendEditingController.php b/src/Controller/FrontendEditingController.php
index 6e74d5c..c3dd797 100644
--- a/src/Controller/FrontendEditingController.php
+++ b/src/Controller/FrontendEditingController.php
@@ -445,7 +445,12 @@ class FrontendEditingController extends ControllerBase {
     if ($message) {
       $response->addCommand(new MessageCommand($message, NULL, ['type' => 'error']));
     }
-    $updated_content = $paragraph->getParentEntity()->get($paragraph->get('parent_field_name')->value)->view($view_mode_id);
+
+    $updated_content_field = $paragraph->getParentEntity()->get($paragraph->get('parent_field_name')->value);
+    $updated_content = $view_mode_id === '_custom'
+      ? $updated_content_field->view()
+      : $updated_content_field->view($view_mode_id);
+
     foreach (Element::getVisibleChildren($updated_content) as $delta) {
       $item = $updated_content[$delta];
       if (!empty($item['#pre_render'])) {
